File: 프로그래머스/unrated/181890. 왼쪽 오른쪽/왼쪽 오른쪽.java
class Solution {
    public String[] solution(String[] str_list) {
        String[] answer = {};
        
        int s = 0;
        boolean l = false, r = false;
        for(int i = 0; i < str_list.length; i++) {
            if(str_list[i].equals("l") || str_list[i].equals("r")) {
                if(str_list[i].equals("l")) l = true;
                else r = true;
                s = i;
                break;
            }
        }
        
        if(l) {
            answer = new String[s];
            for(int i = 0; i < s; i++) {
                answer[i] = str_list[i];
            }
        }
        
        else if(r) {
            int len = str_list.length - s - 1;
            answer = new String[len];
            for(int i = s + 1; i < str_list.length; i++) {
                answer[i - s - 1] = str_list[i];
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.js
function solution(str_list) {
    let answer = [];
    
    let idxOfL = str_list.indexOf('l');
    let idxOfR = str_list.indexOf('r');
    
    if(idxOfL + idxOfR == -2) return answer;
    
    if(idxOfL == -1) {
        for(let i = idxOfR + 1; i < str_list.length; i++) {
            answer.push(str_list[i]);
        }
        return answer;
    }
    if(idxOfR == -1) {
        for(let i = 0; i < idxOfL; i++) {
            answer.push(str_list[i]);
        }
        return answer;
    }
    if(idxOfL < idxOfR) {
        for(let i = 0; i < idxOfL; i++) {
            answer.push(str_list[i]);
        }
        return answer;
    }
    if(idxOfR < idxOfL) {
        for(let i = idxOfR + 1; i < str_list.length; i++) {
            answer.push(str_list[i]);
        }
        return answer;
    }
    
    return answer;
}